src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -38,10 +111,168 @@ private void SetContentDispositionHeader(ActionContext context, FileResult resul
             }
         }
 
-        private void SetContentType(ActionContext context, FileResult result)
+        private static void SetAcceptRangeHeader(ActionContext context)
         {
             var response = context.HttpContext.Response;
-            response.ContentType = result.ContentType;
+            response.Headers[HeaderNames.AcceptRanges] = AcceptRangeHeaderValue;
+        }
+
+        // Internal for testing
+        internal static PreconditionState GetPreconditionState(
```

Comment:
Hmm, that RFC was released about the time the original StaticFiles code was written...
Giving PreconditionFailed the highest priority has almost the same effect as that stated ordering.